plugins/outputs/stackdriver/stackdriver.go

```diff
@@ -0,0 +1,293 @@
+	QuotaLabelsPerMetricDescriptor = 10
+	// QuotaStringLengthForLabelKey is the limit
+	// to string length for label key.
+	QuotaStringLengthForLabelKey = 100
+	// QuotaStringLengthForLabelValue is the limit
+	// to string length for label value.
+	QuotaStringLengthForLabelValue = 1024
+
+	// StartTime for cumulative metrics.
+	StartTime = int64(1)
+	// MaxInt is the max int64 value.
+	MaxInt = int(^uint(0) >> 1)
+)
+
+var sampleConfig = `
+  # GCP Project
+  project = "erudite-bloom-151019"
+
+  # The namespace for the metric descriptor
+  namespace = "telegraf"
+`
+
+// Connect initiates the primary connection to the GCP project.
+func (s *Stackdriver) Connect() error {
+	if s.Project == "" {
+		return fmt.Errorf("Project is a required field for stackdriver output")
+	}
+
+	if s.Namespace == "" {
+		return fmt.Errorf("Namespace is a required field for stackdriver output")
+	}
+
+	if s.client == nil {
+		ctx := context.Background()
+		client, err := monitoring.NewMetricClient(ctx)
+		if err != nil {
+			return err
+		}
+		s.client = client
+	}
+
+	return nil
+}
+
+// Write the metrics to Google Cloud Stackdriver.
+func (s *Stackdriver) Write(metrics []telegraf.Metric) error {
+	ctx := context.Background()
+
+	for _, m := range metrics {
+		timeSeries := []*monitoringpb.TimeSeries{}
+
+		for k, v := range m.Fields() {
+			value, err := getStackdriverTypedValue(v)
+			if err != nil {
+				log.Printf("E! [output.stackdriver] get type failed: %s", err)
+				continue
+			}
+
+			metricKind, err := getStackdriverMetricKind(m.Type())
+			if err != nil {
+				log.Printf("E! [output.stackdriver] get metric failed: %s", err)
+				continue
+			}
+
+			timeInterval, err := getStackdriverTimeInterval(metricKind, StartTime, m.Time().Unix())
+			if err != nil {
+				log.Printf("E! [output.stackdriver] get time interval failed: %s", err)
+				continue
+			}
+
+			// Prepare an individual data point.
+			dataPoint := &monitoringpb.Point{
+				Interval: timeInterval,
+				Value:    value,
+			}
+
+			// Prepare time series.
+			timeSeries = append(timeSeries,
+				&monitoringpb.TimeSeries{
+					Metric: &metricpb.Metric{
+						Type:   path.Join("custom.googleapis.com", s.Namespace, m.Name(), k),
+						Labels: getStackdriverLabels(m.Tags()),
+					},
+					MetricKind: metricKind,
+					Resource: &monitoredrespb.MonitoredResource{
+						Type: "global",
+						Labels: map[string]string{
+							"project_id": s.Project,
+						},
+					},
+					Points: []*monitoringpb.Point{
+						dataPoint,
+					},
+				})
+		}
+
+		if len(timeSeries) < 1 {
+			continue
+		}
+
+		// Prepare time series request.
+		timeSeriesRequest := &monitoringpb.CreateTimeSeriesRequest{
+			Name:       monitoring.MetricProjectPath(s.Project),
+			TimeSeries: timeSeries,
+		}
+
+		// Create the time series in Stackdriver.
+		err := s.client.CreateTimeSeries(ctx, timeSeriesRequest)
+		if err != nil {
+			log.Printf("E! [output.stackdriver] create time series failed: %s", err)
```

Comment:
I know we talked about this earlier, but I didn't really get a sense of why we don't return an error here.  If we return an error then Telegraf will treat all the data as not written and will try it again on the next `flush_interval`, without an error the data will be lost.